opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -199,16 +201,16 @@ def export(self) -> None:
             else:
                 self.spans_list[idx] = span
                 idx += 1
-        with Context.use(suppress_instrumentation=True):
-            try:
-                # Ignore type b/c the Optional[None]+slicing is too "clever"
-                # for mypy
-                self.span_exporter.export(
-                    self.spans_list[:idx]
-                )  # type: ignore
-            # pylint: disable=broad-except
-            except Exception:
-                logger.exception("Exception while exporting Span batch.")
+        backup_context = get_current()
```

Comment:
It's an overkill IMHO, notice that the previous implementation was only setting the value of the arguments passed in `use()` instead of creating a full copy of the context.

https://github.com/open-telemetry/opentelemetry-python/blob/7229435e1d86679c5e9df380ddcafe2171efdb44/opentelemetry-api/src/opentelemetry/context/base_context.py#L104-L110

Reply:
But it could do that because each separate value was a contextvar under the hood (at least in the async case):

https://github.com/open-telemetry/opentelemetry-python/blob/7229435e1d86679c5e9df380ddcafe2171efdb44/opentelemetry-api/src/opentelemetry/context/async_context.py#L44-L45

With everything in a single `Context` I think we have no choice but to make a copy.

Reply:
Yes, you're right and that's exactly my point. Having an immutable `Context` avoids to worry about concurrency executions accessing the same `Context` object. I don't know if that also makes sense when the context is propagated automatically, in this case there is a different **implicit** context for each task (provided under the hood by contextvars/threadlocal) and the concurrency access is not an issue anymore.
I already talked to @codeboten and we agree that the best idea is to move ahead and explore this later on.